cppcache/src/CacheImpl.cpp

```diff
@@ -46,15 +45,15 @@
 
 using namespace apache::geode::client;
 
-CacheImpl::CacheImpl(Cache* c, std::unique_ptr<DistributedSystem> sys,
-                     bool iPUF, bool readPdxSerialized,
+CacheImpl::CacheImpl(Cache* c, DistributedSystem distributedSystem, bool iPUF,
```

Comment:
You have changed this method from taking ownership of `DistributedSystem` to making a copy. This method should be taking an r-value and moving ownership to the `CacheImpl`.